variables.tf

```diff
@@ -177,170 +200,372 @@ variable "additional_node_groups" {
+    install_binaries         = optional(bool)
   })
-  description = "if specifed, a bastion is created with the specified details"
-  default     = null
+  default = {}
 }
 
-variable "efs_access_point_path" {
-  type        = string
-  description = "Filesystem path for efs."
-  default     = "/domino"
-}
 
-variable "ssh_pvt_key_path" {
-  type        = string
-  description = "SSH private key filepath."
-  validation {
-    condition     = fileexists(var.ssh_pvt_key_path)
-    error_message = "Private key does not exist. Please provide the right path or generate a key with the following command: ssh-keygen -q -P '' -t rsa -b 4096 -m PEM -f domino.pem"
+variable "storage" {
+  description = <<EOF
+    storage = {
+      efs = {
+        access_point_path = "Filesystem path for efs."
+        backup_vault = {
+          create        = "Create backup vault for EFS toggle."
+          force_destroy = "Toggle to allow automatic destruction of all backups when destroying."
+          backup = {
+            schedule           = optional(string)
+            cold_storage_after = "Move backup data to cold storage after this many days."
+            delete_after       = "Delete backup data after this many days."
+          }
+        }
+      }
+      s3 = {
+        force_destroy_on_deletion = "Toogle to allow recursive deletion of all objects in the s3 buckets. if 'false' terraform will NOT be able to delete non-empty buckets"
+      }
+      ecr = {
+        force_destroy_on_deletion = "Toogle to allow recursive deletion of all objects in the ECR repositories. if 'false' terraform will NOT be able to delete non-empty repositories"
+      }
+    }
   }
+  EOF
+  type = object({
+    efs = optional(object({
+      access_point_path = optional(string)
+      backup_vault = optional(object({
+        create        = optional(bool)
+        force_destroy = optional(bool)
+        backup = optional(object({
+          schedule           = optional(string)
+          cold_storage_after = optional(number)
+          delete_after       = optional(number)
+        }))
+      }))
+    }))
+    s3 = optional(object({
+      force_destroy_on_deletion = optional(bool)
+    }))
+    ecr = optional(object({
+      force_destroy_on_deletion = optional(bool)
+    }))
+  })
+  default = {}
 }
 
-variable "s3_force_destroy_on_deletion" {
-  description = "Toogle to allow recursive deletion of all objects in the s3 buckets. if 'false' terraform will NOT be able to delete non-empty buckets"
-  type        = bool
-  default     = false
-}
-
-variable "ecr_force_destroy_on_deletion" {
-  description = "Toogle to allow recursive deletion of all objects in the ECR repositories. if 'false' terraform will NOT be able to delete non-empty repositories"
-  type        = bool
-  default     = false
-}
+variable "kms" {
+  description = <<EOF
+    enabled = "Toggle,if set use either the specified KMS key_id or a Domino-generated one"
```

Comment:
I don't understand. if they gave you a key, we should use it otherwise its domino managed. Why do we need this enabled flag?

Reply:
Meant to replace the current `var.use_kms`

Reply:
I dont think we meant to enforce `kms` given default is `false` 
https://github.com/dominodatalab/terraform-aws-eks/blob/main/variables.tf#L294